File: AnimationObjeC/AnimationObjeC/ViewController/AnimationViewController.m
//
//  AnimationViewController.m
//  AnimationObjeC
//
//  Created by Chitralekha Yellewar on 21/08/18.
//  Copyright © 2018 Chitralekha Yellewar. All rights reserved.
//

#import "AnimationViewController.h"
#import "Constants.h"

@interface AnimationViewController ()
    
@property (weak, nonatomic) IBOutlet UILabel *subTitleLabel;
@property (weak, nonatomic) IBOutlet UILabel *titleLabel;
@property (weak, nonatomic) IBOutlet NSLayoutConstraint *titleBottomConstraint;
    
@end

@implementation AnimationViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self hideLabels];
    [self startAnimation];
}
    
- (void)hideLabels {
    self.titleLabel.alpha = ALPHA_ZERO;
    self.subTitleLabel.alpha = ALPHA_ZERO;
}
    
- (void)startAnimation {
    [self animateTitleLabel];
}

#pragma Mark- animate title label
    
- (void)animateTitleLabel {
    [UIView animateWithDuration:ANIMATE_DURATION delay:ANIMATE_DELAY options:UIViewAnimationOptionTransitionCurlUp animations:^{
        self.titleLabel.alpha = ALPHA_ONE;
        [self.titleBottomConstraint setConstant:self.view.frame.size.height * BOTTOM_MULTIPLIER];
        [self.view layoutIfNeeded];
    } completion:^(BOOL finished) {
        [self animateSubtitleLabel];
    }];
}

#pragma Mark- animate subtitle label
    
- (void)animateSubtitleLabel {
    [UIView animateWithDuration:ANIMATE_DURATION delay:ANIMATE_DELAY options:UIViewAnimationOptionTransitionCurlUp animations:^{
        self.subTitleLabel.alpha = ALPHA_ONE;
        [self.view layoutIfNeeded];
    } completion:^(BOOL finished) {
        NSLog(@"Animation complete !!");
    }];
}
    
@end

File: Animation/Animation/ViewController/AnimationViewController.swift
//
//  AnimationViewController.swift
//  Animation
//
//  Created by Chitralekha Yellewar on 22/08/18.
//  Copyright © 2018 Chitralekha Yellewar. All rights reserved.
//

import UIKit

class AnimationViewController: UIViewController {
    
    @IBOutlet weak var titleLabel: UILabel!
    @IBOutlet weak var titleBottomContraint: NSLayoutConstraint!
    @IBOutlet weak var subtitleLabel: UILabel!
    let constants = Constants()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        hideLabels()
        startAnimation()
    }
    
    func hideLabels() {
        self.titleLabel.alpha = .zeroAlpha
        self.subtitleLabel.alpha = .zeroAlpha
    }
    
    func startAnimation() {
        animateTitleLabel()
    }
    
    // MARK: animate title.
    func animateTitleLabel() {
        UIView.transition(with: self.titleLabel, duration: TimeInterval(constants.ANIMATE_DURATION), options: UIViewAnimationOptions.curveEaseIn, animations: {
            self.titleLabel.alpha = .oneAplha
            self.titleBottomContraint.constant = self.view.frame.height*CGFloat.labelBottomMutliplier
            self.view.layoutIfNeeded()
        }, completion: { (finished) in
            self.animateSubtitle()
        })
    }
    
    //MARK: animate subtitle.
    func animateSubtitle() {
        UIView.animate(withDuration: TimeInterval(constants.ANIMATE_DURATION), animations: {
            self.subtitleLabel.alpha = .oneAplha
            self.view.layoutIfNeeded()
            print("Animation Done")
            })
    }
}

extension CGFloat {
    public static let zeroAlpha = CGFloat(0.0)
    public static let oneAplha = CGFloat(1.0)
    public static let labelBottomMutliplier = CGFloat(0.55)
}
